schema/dsl/parse_test.go

```diff
@@ -1,27 +1,79 @@
 package schemadsl_test
 
 import (
+	"bytes"
+	"flag"
+	"io/ioutil"
 	"testing"
 
+	ipldjson "github.com/ipld/go-ipld-prime/codec/json"
+	"github.com/ipld/go-ipld-prime/node/bindnode"
 	"github.com/ipld/go-ipld-prime/schema"
 	schemadmt "github.com/ipld/go-ipld-prime/schema/dmt"
 	schemadsl "github.com/ipld/go-ipld-prime/schema/dsl"
+
+	qt "github.com/frankban/quicktest"
 )
 
-func TestParse(t *testing.T) {
-	sch, err := schemadsl.ParseFile("../../.ipld/specs/schemas/schema-schema.ipldsch")
-	if err != nil {
-		t.Fatal(err)
-	}
+var update = flag.Bool("update", false, "update testdata files in-place")
+
+func TestParseSchemaSchema(t *testing.T) {
+	t.Parallel()
+
+	inputSchema := "../../.ipld/specs/schemas/schema-schema.ipldsch"
+	inputJSON := "../../.ipld/specs/schemas/schema-schema.ipldsch.json"
+
+	src, err := ioutil.ReadFile(inputSchema)
+	qt.Assert(t, err, qt.IsNil)
+
+	srcJSON, err := ioutil.ReadFile(inputJSON)
+	qt.Assert(t, err, qt.IsNil)
+
+	testParse(t, string(src), string(srcJSON), func(updated string) {
+		err := ioutil.WriteFile(inputJSON, []byte(updated), 0o777)
+		qt.Assert(t, err, qt.IsNil)
+	})
+}
+
+func testParse(t *testing.T, inSchema, inJSON string, updateFn func(string)) {
```

Comment:
FYI, I wrote this helper because the intent is to add a table-driven tests to reuse @rvagg's previous test input suite at https://github.com/ipld/go-ipld-schema/tree/master/test/fixtures/bulk

Reply:
Or that :) I'll probably do that as part of the next PR.